### PR TITLE
Feat(scoped variables): file view

### PR DESCRIPTION
**Pull request by AbhishekA1509**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

This PR introduces layout for variables when user already has uploaded a file, it has following functionalities:
* User can re-upload a file and it will show the progress bar, then upload-editor-view and then the same view.
* User can select to download file or template through dropdown.
* User should be able to toggle tabs.

Snapshots:
Screenshot 2023-08-22 at 2 36 01 PM

Dep: https://github.com/devtron-labs/dashboard/pull/1222

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/scopedVariables/types.tsx [hunk 1]
```diff
@@ -5,6 +5,16 @@ export enum ReadFileAs {
     ARRAY_BUFFER = 'arrayBuffer',
 }
 
+export enum FileView {
```
Comment (shivani170) [inline src/components/scopedVariables/types.tsx:8]: Can optimize ConfigurationType from constant

File: src/components/scopedVariables/types.tsx [hunk 2]
```diff
@@ -5,6 +5,16 @@ export enum ReadFileAs {
...
+    YAML = 'yaml',
+    SAVED = 'saved',
+}
+
+export interface SavedVariablesViewI {
+    scopedVariables: any
```
Comment (shivani170) [inline src/components/scopedVariables/types.tsx:14]: Please define the type if possible

File: src/components/scopedVariables/types.tsx [hunk 3]
```diff
@@ -5,6 +5,16 @@ export enum ReadFileAs {
...
+    setScopedVariables: (variables: any) => void
```
Comment (shivani170) [inline src/components/scopedVariables/types.tsx:15]: Variables type?

File: src/components/scopedVariables/types.tsx (hunk 2 above)
Comment (AbhishekA1509) [reply src/components/scopedVariables/types.tsx:14]: Right now schema, is subject to change and variables can have any value.

File: src/components/scopedVariables/types.tsx (hunk 3 above)
Comment (AbhishekA1509) [reply src/components/scopedVariables/types.tsx:15]: Right now the schema is not fixed, will update it on confirmation.

File: src/components/scopedVariables/types.tsx (hunk 1 above)
Comment (AbhishekA1509) [reply src/components/scopedVariables/types.tsx:8]: Actually this enum is supposed to restrict the type of state to two views only, so for better separation of concerns I haven't used this.

**Comment by AbhishekA1509** [comment]
> Please reuse ic-upload in place of ic-upload-blue

Done replaced ic-upload with ic-upload-blue

**Pull request closed without merging**